miio/g1vacuum.py

```diff
@@ -0,0 +1,281 @@
+    "reset_filter_life_level":  {"siid": 11, "aiid": 1}
+    }
+}
+
+error_codes = {
+    0: "No error",
+    1: "Left Wheel stuck",
+    2: "Right Wheel stuck",
+    3: "Cliff error",
+    4: "Low battery",
+    5: "Bump error",
+    6: "Main Brush Error",
+    7: "Side Brush Error",
+    8: "Fan Motor Error",
+    9: "Dustbin Error",
+    10: "Charging Error",
+    11: "No Water Error",
+    12: "Pick Up Error"
+}
+
+class G1ChargeState(Enum):
+    Discharging = 0
+    Charging = 1
+    FullyCharged = 2
+
+class G1State(Enum):
+    """Vacuum Status"""
+    Idle = 1
+    Sweeping = 2
+    Paused = 3
+    Error = 4
+    Charging = 5
+    GoCharging = 6
+
+class G1Consumable(Enum):
+    """Consumables"""
+    MainBrush = "main_brush_life_level"
+    SideBrush = "side_brush_life_level"
+    Filter = "filter_life_level"
+
+class G1VacuumMode(Enum):
+    """Vacuum Mode"""
+    GlobalClean = 1
+    SpotClean = 2
+    Wiping = 3
+
+class G1WaterLevel(Enum):
+    """Water Flow Level"""
+    Level1 = 1
+    Level2 = 2
+    Level3 = 3
+
+class G1FanSpeed(Enum):
+    """Fan speeds, same as for ViomiVacuum."""
+    Mute = 0
+    Standard = 1
+    Medium = 2
+    High = 3
+
+class G1Languages(Enum):
+    Chinese = 0
+    English = 1
+
+class G1MopState(Enum):
+    Off = 0
+    On = 1
+
+
+class G1Status(DeviceStatus):
+    """Container for status reports from the Mijia Vacuum G1."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.data = data
+
+    @property
+    def battery(self) -> int:
+        """Battery Level."""
+        return self.data["battery"]
+
+    @property
+    def charge_state(self) -> G1ChargeState:
+        """Charging State."""
+        return G1ChargeState[G1ChargeState(self.data["charge_state"]).name]
+
+    @property
+    def error_code(self) -> int:
+        """Error code as returned by the device."""
+        return int(self.data["error_code"])
+
+    @property
+    def error(self) -> str:
+        """Human readable error description, see also :func:`error_code`."""
+        try:
+            return error_codes[self.error_code]
+        except KeyError:
+            return "Definition missing for error %s" % self.error_code
+
+    @property
+    def state(self) -> G1State:
+        """Vacuum Status."""
+        return G1State[G1State(self.data["state"]).name]
+
+    @property
+    def fan_speed(self) -> G1FanSpeed:
+        """Fan Speed."""
+        return G1FanSpeed[G1FanSpeed(self.data["fan_speed"]).name]
+
+    @property
+    def operating_mode(self) -> G1VacuumMode:
+        """Operating Mode."""
+        return G1VacuumMode[G1VacuumMode(self.data["operating_mode"]).name]
+
+    @property
+    def mop_state(self) -> G1MopState:
+        """Mop State."""
+        return G1MopState[G1MopState(self.data["mop_state"]).name]
+
+    @property
+    def water_level(self) -> G1WaterLevel:
+        """Mop State."""
+        return G1WaterLevel[G1WaterLevel(self.data["water_level"]).name]
```

Comment:
```suggestion
        return G1WaterLevel(self.data["water_level"])
```
There is no need to create an enum to read its name to create the presentation of the same enum :-) Please also convert the other enum states to avoid doing that.